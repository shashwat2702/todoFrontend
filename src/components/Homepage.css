.homePageContainer {
    display: flex;
    flex-direction: column;
    flex: 1;
    padding: 2rem;
    min-width: 700px;
    align-items: center;
}
.addTaskContainer {
    width: 100%;
}
.addTodoInputField {
    background-color: rgb(30, 31, 31);
    border: 0;
    border-bottom: 2px solid #9a98ad;
    width: 100%;
    height: 3rem;
    font-size: 2rem;
    color: white;
    margin-top: 4rem;
    outline-color: antiquewhite;
    padding: 0.5rem;
}
.addTaskButton {
    background-color: dodgerblue;
    width: 10rem;
    height: 4rem;
    border: 0;
    border-radius: 0.2rem;
    margin-top: 1rem;
    font-size: 2rem;
    color: white;
}
.listOfTasks {
    margin: 2rem;
    border: 2px solid whitesmoke;
    border-radius: 0.5rem;
    min-height: 400px;
    padding: 2rem;
    width: 100%;
    color: white;
    width: 100%;
    font-size: 2rem; 
}