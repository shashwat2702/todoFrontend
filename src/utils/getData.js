const axios = require('axios');

const getData = url => axios.get(url);
const postData = (url, data) => axios({
  method: 'post',
  url,
  data,
});
export { getData, postData };
