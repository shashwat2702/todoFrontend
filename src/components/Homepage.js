import React, { Component } from 'react';
import './Homepage.css';

export default class Homepage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: '',
    };
  }

  handleInputChange = (event) => {
    const { target } = event;
    const { value } = target;
    const { name } = target;
    this.setState({
      [name]: value,
    });
  }

  render() {
    const { message } = this.state;
    return (
      <div className="homePageContainer">
        <input
          type="text"
          name="message"
          value={message}
          onChange={this.handleInputChange}
        />
      </div>
    );
  }
}
