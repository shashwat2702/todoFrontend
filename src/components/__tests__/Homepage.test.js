import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Homepage from '../Homepage';
import * as apiCalls from '../../utils/getData';

describe('Homepage snapshot', () => {
  it('should render correctly', () => {
    const HomepageSnap = renderer.create(<Homepage />);
    const HomepageJson = HomepageSnap.toJSON();
    expect(HomepageJson).toMatchSnapshot();
  });
});

describe('getTodoTasks', () => {
  it('should call getData', () => {
    jest.spyOn(apiCalls, 'getData');
    const wrapper = shallow(<Homepage />);
    const instance = wrapper.instance();
    instance.getTodoTasks();
    expect(apiCalls.getData).toHaveBeenCalled();
  });
});

describe('handleInputChange', () => {
  it('should set value in newTask attribute of state', () => {
    const wrapper = shallow(<Homepage />);
    const instance = wrapper.instance();
    const event = {
      target: {
        name: 'newTask',
        value: 'Create',
      },
    };
    instance.handleInputChange(event);
    expect(instance.state.newTask).toEqual(event.target.value);
  });
});

describe('addTask', () => {
  it('should not call postData when newTask is empty', () => {
    jest.spyOn(apiCalls, 'postData');
    const wrapper = shallow(<Homepage />);
    const instance = wrapper.instance();
    instance.addTask({ preventDefault: jest.fn() });
    expect(apiCalls.postData).not.toHaveBeenCalled();
  });
  it('should call postData when newTask has some data', () => {
    jest.spyOn(apiCalls, 'postData');
    const wrapper = shallow(<Homepage />);
    const instance = wrapper.instance();
    instance.setState({ newTask: 'Create Todo App' });
    instance.addTask({ preventDefault: jest.fn() });
    expect(apiCalls.postData).toHaveBeenCalled();
  });
});

describe('deleteTask', () => {
  it('should call deleteTask with parameter id', () => {
    jest.spyOn(apiCalls, 'deleteData');
    const wrapper = shallow(<Homepage />);
    const event = {
      target: {
        id: 1,
      },
    };
    const instance = wrapper.instance();
    instance.deleteTask(event);
    expect(apiCalls.deleteData).toHaveBeenCalledWith('http://localhost:8080/removeTask', event.target.id);
  });
});
