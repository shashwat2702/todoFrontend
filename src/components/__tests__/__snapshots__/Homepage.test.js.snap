// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage snapshot should render correctly 1`] = `
<div
  className="homePageContainer"
>
  <form
    className="addTaskContainer"
    onSubmit={[Function]}
  >
    <input
      className="addTodoInputField"
      name="newTask"
      onChange={[Function]}
      placeholder="What needs to be done?"
      type="text"
      value=""
    />
    <button
      className="addTaskButton"
      onClick={[Function]}
      type="submit"
    >
      Add Task
    </button>
  </form>
  <div
    className="headingTodo"
  >
    ALL THE TODO TASKS
  </div>
  <div
    className="listOfTasks"
  />
</div>
`;
